dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java

```diff
@@ -1,126 +1,129 @@
+                .getInt("test")>0;
 
-		int test = db.getInt("test");
-		return (test > 0);
 	}
 
-	@CloseDBIfOpened
-	public static void InitializeDb() throws DotDataException {
-
-		final boolean configured = isConfigured();
-
-		if (!configured) {
-			if(Config.getBooleanProperty("STARTERSITE_BUILD", true)){
-				Logger.info(DotCMSInitDb.class,"There are no inodes - initializing db with starter site");
-				try {
-					PrintWriter pw = new PrintWriter(new StringWriter());
-					loadStarterSite(pw);
-					Logger.info(DotCMSInitDb.class, pw.toString());
-				} catch (IOException e) {
-					Logger.error(DotCMSInitDb.class, "Unable to load starter site", e);
-				}
-			}else{
-				Logger.info(DotCMSInitDb.class,"There are no inodes - initializing db for first time use");
-				buildDefaultData();
-			}
-		}else {
-			Logger.info(DotCMSInitDb.class,"inodes exist, skipping initialization of db");
-		}
-	}
+    @CloseDBIfOpened
+    public static void InitializeDb() {
 
-	private static void buildDefaultData() throws DotDataException {
-		try {
-			HostAPI hostAPI = APILocator.getHostAPI();
-			LocalTransaction.wrap(() -> PublicCompanyFactory.createDefaultCompany());
-			// Ensures that default groups are set up
-	//		GroupFactory.createDefaultGroups();
-	//		try {
-	//			LayoutFactory.createDefaultLayouts();
-	//		} catch (Exception e) {
-	//			throw new DotDataException(e.getMessage(), e);
-	//		}
-			// Creating the default host
-			User systemUser = APILocator.getUserAPI().getSystemUser();
-			List<Host> hosts = hostAPI.findAll(systemUser, false);
-			if (hosts.size() == 0) {
-				Logger.debug(DotCMSInitDb.class, "Creating Default Host");
-				hostAPI.findDefaultHost(systemUser, false);
-			}
-	        //Create Default Language
-	        APILocator.getLanguageAPI().createDefaultLanguage();
-			// Creating the default content structures if it not exists.
-			StructureFactory.createDefaultStructure();
-		} catch (DotSecurityException e) {
-			Logger.fatal(DotCMSInitDb.class, "Unable to initialize default data", e);
-			throw new DotRuntimeException(e.getMessage(), e);
-		} catch (Exception e) {
-			Logger.fatal(DotCMSInitDb.class, "Unable to initialize default data", e);
-			new DotDataException(e.getMessage(), e);
-		}
-	}
+        if (!isConfigured()) {
 
-	private static void loadStarterSite(PrintWriter pw) throws IOException{
-		
-		String starter = Config.getStringProperty("STARTER_DATA_LOAD");
-		File starterZip = null;
-		
-		if(UtilMethods.isSet(starter)){
+            Logger.info(DotCMSInitDb.class, "There are no inodes - initializing db with starter site");
 
-			// First we try using the real path
+    
+            
+            Try.run(() -> loadStarterSite()).getOrElseThrow(e->new DotRuntimeException(e));
+
+
+        } else {
+            Logger.info(DotCMSInitDb.class, "inodes exist, skipping initialization of db");
+        }
+    }
+
+    
+    @WrapInTransaction
+    private static void loadStarterSiteData() throws Exception{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)